ts/parsers/graphicFrameParser.ts

```diff
@@ -0,0 +1,74 @@
+//Graphic frame node includes tables, charts and diagrams
+
+import { getAttributeByPath } from "../helpers/attributesHandler";
+
+export default class GraphicFrameParser {
+  public static processGraphicFrameNodes = (graphicFrames) => {
+    const result = [];
+    for (const frame of graphicFrames) {
+      const graphicTypeUri = getAttributeByPath(
+        [frame],
+        ["a:graphic", "a:graphicData", "$"]
+      ).uri;
+      switch (graphicTypeUri) {
+        case "http://schemas.openxmlformats.org/drawingml/2006/table":
+          result.push(frame);
+          break;
+        case "http://schemas.openxmlformats.org/drawingml/2006/chart":
+          break;
+        case "http://schemas.openxmlformats.org/drawingml/2006/diagram":
+          break;
+        default:
+      }
+    }
+
+    return result;
+  };
+
+  public static extractTableElements = (frame) => {
+    const rawTable = getAttributeByPath(
+      [frame],
+      ["a:graphic", "a:graphicData", "a:tbl"]
+    );
+    const rawRows = rawTable[0]["a:tr"] ? rawTable[0]["a:tr"] : [];
+
+    const tableRows = rawRows.map((row) => {
+      let cols = row["a:tc"] ? row["a:tc"] : [];
+      cols = cols.filter(col => {
+        //filtering the columns that are merge columns or merge rows. as we still get them in raw data
+        if ( col['$'] && (col['$']['vMerge'] || col['$']['hMerge'])) {
+            return false;
+        }
+        return true;
+      });
+
+      cols = cols.map((col) => {
+        let meta = {};
+        if (col['$']) {
+            if ((col['$']['rowSpan'])) {
+                meta['rowSpan'] = col['$']['rowSpan'];
+            }
+            if ((col['$']['gridSpan'])) {
+                meta['colSpan'] = col['$']['gridSpan'];
+            }
+        }
+        //can a:t have more values ? more than one in the array
+        //return any other column info, eg width height?
```

Comment:
is this a todo? mark it as such so you can find it later, if not, then perhaps see if you can put an answer as a comment rather a question

Reply:
made it a todo